File: activities/07-tree/Python/TreeNode.py
#TreeNode:top#
#TreeNode:import#

# <tree>:TreeNode ::= LPAREN <TAG> <tree>:leftNode <tree>:rightNode RPAREN
class TreeNode(Tree): #TreeNode:class#

    className = "TreeNode"
    ruleString = "<tree>:TreeNode ::= LPAREN <TAG> <tree>:leftNode <tree>:rightNode RPAREN"
    tag = None
    leftNode = None
    rightNode = None

    def __init__(tag, leftNode, rightNode):
        #TreeNode:init#
        self.tag = tag
        self.leftNode = leftNode
        self.rightNode = rightNode

#TreeNode#

File: activities/07-tree/Java/TreeNode.java
//TreeNode:top//
//TreeNode:import//
import java.util.*;

// <tree>:TreeNode ::= LPAREN <TAG> <tree>:leftNode <tree>:rightNode RPAREN
public class TreeNode extends Tree /*TreeNode:class*/ {

    public static final String $className = "TreeNode";
    public static final String $ruleString =
        "<tree>:TreeNode ::= LPAREN <TAG> <tree>:leftNode <tree>:rightNode RPAREN";

    public Token tag;
    public Tree leftNode;
    public Tree rightNode;

    public TreeNode(Token tag, Tree leftNode, Tree rightNode) {
//TreeNode:init//
        this.tag = tag;
        this.leftNode = leftNode;
        this.rightNode = rightNode;
    }

    public static TreeNode parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<tree>:TreeNode", scn$.lno);
        scn$.match(Token.Match.LPAREN, trace$);
        Token tag = scn$.match(Token.Match.TAG, trace$);
        Tree leftNode = Tree.parse(scn$, trace$);
        Tree rightNode = Tree.parse(scn$, trace$);
        scn$.match(Token.Match.RPAREN, trace$);
        return new TreeNode(tag, leftNode, rightNode);
    }

    public String toString() {
        return "(" + tag + " " + leftNode + " " + rightNode + ")";
    }

//TreeNode//
}
